azure/scope/managedcontrolplane.go

```diff
@@ -422,6 +422,7 @@ func (s *ManagedControlPlaneScope) IsVnetManaged() bool {
 	defer done()
 
 	vnet := s.VNetSpec().ResourceRef()
+	vnet.SetNamespace(s.ASOOwner().GetNamespace())
```

Comment:
Why is the namespace set here explicitly? I see that other resources have the namespace set in the [generic aso resource](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4499/files#diff-1bab5ce3b3ab8fd54586c39ef9f40f1c206bfc351a580a5c50fa5c464499bd61R83), but I could be misunderstanding how that works.

Reply:
The comment on the interface definition for `ResourceRef()` now explains that it no longer reads the namespace from there and instead uses the namespace of the owner. This is to account for that change. On the plus side, we won't have to add a namespace field to the `Spec` types for all the services we have yet to convert to ASO.